### PR TITLE
Add field names to types in TypedTree

### PR DESCRIPTION
**Pull request by ErikSchierboom**
As suggested in the [Learn me some F# compiler video](https://www.youtube.com/watch?v=-dKf15xSWPY&t=5717s&ab_channel=F%23SoftwareFoundation), it would be beneficial to have field names added to discriminated union fields. This PR is a first attempt at doing so.

I've tried to add names to a fairly limited number of types, as I wanted to get some feedback on the naming that I've used. I've used the following to decide upon the naming:

- Use the names as specified in the comments. Some DU fields had comments that contained suggestions for the field names, in which case I've used those.
- Do a Find on the type and see what the pattern matched field names are where the type is being pattern matched upon.

I've also tried to use the abbreviations specified in the [Coding standards and idioms](https://github.com/dotnet/fsharp/blob/main/docs/compiler-guide.md#coding-standards-and-idioms) guide where possible, but I don't know exactly when to deviate from that.

Let me know if this PR is headed in the right direction, or if this needs work.

### CODE REVIEW

**Review by vzarytovskii — APPROVED** [review]
That is great. Thank you!

**Comment by ErikSchierboom** [comment]
@vzarytovskii Great to hear! I'll probably do more follow-up PRs then :)

**Review by cartermp — APPROVED** [review]
Thanks, like the name changes.

File: src/fsharp/TypedTree.fs [hunk 1]
```diff
@@ -2359,18 +2363,18 @@ type TraitConstraintSln =
     ///    ty -- the type and its instantiation
     ///    vref -- the method that solves the trait constraint
     ///    minst -- the generic method instantiation 
-    | FSMethSln of TType * ValRef * TypeInst 
+    | FSMethSln of ty: TType * vref: ValRef * minsts: TypeInst
```
Comment (ScottHutchinson) [inline src/fsharp/TypedTree.fs:2366]: minst

File: src/fsharp/TypedTree.fs [hunk 2]
```diff
@@ -4825,7 +4829,7 @@ type ValUseFlag =
     /// a .NET 2.0 constrained call. A constrained call is only used for calls where 
     // the object argument is a value type or generic type, and the call is to a method
     //  on System.Object, System.ValueType, System.Enum or an interface methods.
-    | PossibleConstrainedCall of TType
+    | PossibleConstrainedCall of typ: TType
```
Comment (ScottHutchinson) [inline src/fsharp/TypedTree.fs:4832]: ty

**Review by ScottHutchinson — COMMENTED** [review]
Thanks. Just a couple minor quibbles.

**Comment by ErikSchierboom** [comment]
@ScottHutchinson Shall I correct them in a follow-up PR?

**Comment by ScottHutchinson** [comment]
I guess so, since this one has been merged. I don't know how else to do it. Sorry I didn't review this before it was merged. I have no authority here, so do as you think best.

**Comment by ErikSchierboom** [comment]
Will do. I was planning to do a follow-up anyway :)

**Comment by ErikSchierboom** [comment]
@ScottHutchinson I've fixed the issue you mentioned in https://github.com/dotnet/fsharp/pull/10820/files

**Pull request merged**